flytekit/types/structured/structured_dataset.py

```diff
@@ -337,7 +337,7 @@ def to_literal(
     ) -> Literal:
         # If the type signature has the StructuredDataset class, it will, or at least should, also be a
         # StructuredDataset instance.
-        if issubclass(python_type, StructuredDataset):
+        if inspect.isclass(python_type) and issubclass(python_type, StructuredDataset):
```

Comment:
works for python 3.7-3.10 right?

Reply:
yeah, I've tested it with python 3.7~3.10.